python_modules/libraries/dagster-aws/dagster_aws/sfn/sfn_launcher.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import time
+from functools import cached_property
+from typing import TYPE_CHECKING, Optional
+
+import boto3
+from botocore.exceptions import ClientError
+from dagster import (
+    Field,
+    StringSource,
+    _check as check,
+)
+from dagster._core.errors import DagsterExecutionInterruptedError
+from dagster._core.instance import T_DagsterInstance
+from dagster._core.launcher.base import LaunchRunContext, RunLauncher
+from dagster._serdes import ConfigurableClass
+from tenacity import retry, retry_if_exception_type, stop_after_delay, wait_fixed
+
+if TYPE_CHECKING:
+    from dagster._serdes.config_class import ConfigurableClassData
+    from mypy_boto3_stepfunctions import SFNClient
+    from mypy_boto3_stepfunctions.type_defs import (
+        DescribeExecutionOutputTypeDef,
+        StartExecutionInputRequestTypeDef,
+        StopExecutionOutputTypeDef,
+    )
+
+
+SFN_FINISHED_STATUSES = ["FAILED", "SUCCEEDED", "TIMED_OUT", "ABORTED"]
```

Comment:
I would prefer to do it with a waiter instead. See ECS [example](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py#L298).

Reply:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/index.html
Waiter is not implemented for step-functions